### ISSUE TITLE
Some bug reports

### ISSUE THREAD

**Issue by ilightwas** [RS485/LogisticsPipes#885]
Not sure if I should create a bug report for each, decided to put all at once. Sorry for any weird speaking, english is not my first language.

All this was done with logisticspipes-0.9.3.70 in the FTB Infinity evolved expert mode 2.2.2

[1-ItemSink ignoring block metadata](https://www.youtube.com/watch?v=p2QRE8lUwHE)

There's actually 2 things. The first one is minor, the UI showing the modules comes back when you closes it if you are inside a module but with the crafting module it closes completely. Not sure which is the intended way but I do prefer it to close for good like with the crafting modules. That thing coming back every time is annoying.

Now the real bug. ItemSink module don't respect the sub id or metadata, don't know how to call it. That seems to happen when you have 2 chassis pipes in one inventory/machine as I did other tests with only one chassis pipe(Mk3 and Mk5) and a basic pipe.

[2-Modules resetting over swap try](https://www.youtube.com/watch?v=py-jPYOLyoU)

Recipes vanish if you try to swap crafting modules inside the chassis pipe(didn't try with other modules, will do later). 

Would be good if we could swap the modules easily without having to move one to inventory then putting the other.. then grabbing the one in the inventory.. and then putting it.

[3-Pipes ignoring disconnection upgrades on Deep storage](https://www.youtube.com/watch?v=YJkXpuYCnuQ)

Self explanatory in the video, unrouted pipes are connecting too.

4-Logistics request table don't update the inventory

Well this one is kinda hard to reproduce, happens randomly so its hard to record, so no video.

Sometimes when you make multiple request to different items, they arrive at the table but don't update your view of it and you have to re open the table to see them.
A bit annoying, sometimes you think the crafting is taking longer or the item is far but in fact the item might already have arrived.

5-Server times out every player but doesn't crash

This one might be related to issues #881 #877 #871, I have [this log](https://www.dropbox.com/s/aygat0d7s5dhbek/fml-server-1.log?dl=0) that I asked the server admin when it happened one of the times, maybe there's nothing to find but I'm pretty sure the issue comes from LP. Most related with autocraft requests, at random. Also ,(not sure) when you request a craft in a logistics crafting table with the monitoring upgrade, open the crafting tab and other player comes and opens the tab to see the crafting at the same time. 

Another thing that happens often and might have something to do with it is, some crafts in machines, leaves items behind. Say for example ic2 copper cables, the recipe produces 3 copper cables in the metal former, your request only needs 2 cables, it leaves one behind. If you have multiple recipes in one machine that clogs the pipes and items from the subsequente requests keep going back and forth until you manually removes the leftovers. I can say for sure that when the server entered the time out state this was happening.

This is all I could gather as possible things that might trigger the bug.

Again not sure if I should do it here but well... some features request
Logistics Request table improvements
![2015-12-22_20 54 35](https://cloud.githubusercontent.com/assets/6960676/12044660/9a3334f0-ae7c-11e5-94a6-5f39f88f3977.png)

The red button that I highlighted, it's amazing what it does, since you can just get recipes from nei and the system will craft its components, one thing that would make it better its to show in tabs which components are being crafted, like the crafting monitoring shows when you directly request something.

Another thing would be a button that would request the selected item up to 1 stack. A lot of times you have items that you want to request and its annoying to click until you get the exact number, with this say you have 27 chests and you want them all, you select and press this button and it requests all of them, as I said up to one stack.

The last thing I can think for now that some people might already have requested is to have a field to type in the number of items you want instead of being limited to only the + ++ +++ buttons.

**Comment by hron84** [comment]
> The first one is minor, the UI showing the modules comes back when you closes it if you are inside a module but with the crafting module it closes completely. Not sure which is the intended way but I do prefer it to close for good like with the crafting modules. That thing coming back every time is annoying.

No, there is no problem with the current behavior. If you have a Chassis MK2-3-4 pipe, and you would like to install & configure more than one module at once, it would be more annoying to re-open the pipe GUI. The expected behavior is if you close the module GUI go back to the pipe GUI to be able to configure more module(s). If it means you have to hit `ESC` twice... I think it's a good compromise.

On the other aspects, I think you should submit one bug per issue because tracking multiple different issues in different subsystems on one issue is really hard for the developer team. And you have issues with very-very different things.

Also, please inform us with more mod versions (BC, ThermalDynamics if exists, Forge, etc.) since tracking modpack versions is not easy and most modpacks do not provide a development version where developers could debug your issues right.

Thanks for your help

**Comment by bziemons** [comment]
You would normally create an issue for each problem you have, so we can tick them off one by one. It's fine like this for now. Thanks for reporting :+1:
